### PR TITLE
Add the formatOutput method to the Formatter class

### PR DESCRIPTION
**Pull request by gouniLee**
Add the `formatOutput` method to the Formatter class

### CODE REVIEW

File: src/Formatter.js [hunk 1]
```diff
@@ -65,6 +65,17 @@ class Formatter {
      */
     format(result) {
     }
+    /**
+     * Provide the information for the formatter as an object and
+     * return the formatted string that contains a lot of information
+     * according to the formatter's needs.
+     *
+     * @abstract
+     * @param {Object} [options] the options that needs for formatter
```
Comment (ehoogerbeets) [inline src/Formatter.js]: "Information that the method can use to format the output"

Then, add documentation for each property
@param {string} [options.errors] total number of error results found in this run
etc.

File: src/Formatter.js [hunk 2]
```diff
@@ -65,6 +65,17 @@ class Formatter {
      */
     format(result) {
     }
+    /**
+     * Provide the information for the formatter as an object and
+     * return the formatted string that contains a lot of information
+     * according to the formatter's needs.
```
Comment (ehoogerbeets) [inline src/Formatter.js:71]: This needs to describe that the formatOutput call is supposed to format the whole output, including headers, footers, and summaries, and formatted Result instances. If the formatOutput is defined, the linter will not call format directly and it is up to formatOutput to do that. If formatOutput is not defined in a Formatter plugin, then ilib-lint will use its own default headers and footers and will call format to format each Result.

Comment (gouniLee) [reply src/Formatter.js:71]: I update it. Please check it if it's ok.

File: src/Formatter.js [hunk 3]
```diff
@@ -65,6 +65,31 @@ class Formatter {
...
+     * footers and summaries and formatted Result instances.
+     * If the formatOutput is defined, the linter does not call format()
+     * directly and it is up to formatOutput to do so. If formatOutput
+     * is not defined in a Formatter plugin, then ilib-lint will use its
+     * own default headers and footers and will call format to format each Result.
+     *
+     * @abstract
+     * @param {Object} [options] Information that the method can use to format the output
+     * @param {string} [options.name] name of the this project
+     * @param {Object} [options.fileStats] The stats information of the file
```
Comment (ehoogerbeets) [inline src/Formatter.js]: This is a FileStats type

**Review by ehoogerbeets — APPROVED** [review]

**Pull request merged**